pages/rs_quality/fair.md

```diff
@@ -1,5 +1,178 @@
+it is very hard to consider all possible edge cases or notice every single typing mistake. 
+Testing also gives other people confidence in your code as they can see an example of how it is meant to run and be assured that it does work 
+correctly on their machine - helping with code understanding and reusability.
+
+### Coding conventions
+
+Following coding conventions and guides for your programming language that is agreed upon by the community and other programmers
+are important practices to ensure that others find it easy to read your code, reuse or extend it in their own examples and applications.
+
+### Software- and project- level documentation
+
+Documentation comes in many forms - from **software-level documentation** including descriptive names of variables and functions and 
+additional comments that explain lines of your code, to **project-level documentation** (including README, LICENCE, CITATION, CONTRIBUTING, etc. files) 
+that help to discover it, explain the legal terms of reusing it, describe its functionality and how to install, run and contribute to it, 
+to whole websites full of documentation with function definitions, usage examples, tutorials and guides. 
+You many not need as much documentation as a large commercial software product, but making your code reusable relies on other people being able to understand 
+what your code does and how to use it.
+
+### Software repositories and registries
+
+Having somewhere to share your code is fundamental to making it findable and accessible. 
+Your institution might have a code repository, your research field may have a practice of sharing code via a specific website, archive or journal,
+or your version control system might include an online component that makes sharing different versions of your code easy. 
+You should check the rules or guidelines of your institution, grant or domain on publishing code, as well as any licenses of the code your software depends on or reuses.
+
+Some examples of commonly used software repositories and registries include:
+
+- general-purporse software repositories - [GitHub][github] and [GitLab][gitlab]
+- programming language-specific software repositories - [PyPi][pypi] (for Python) and [CRAN][cran] (for R)
+- software registries  - [BioTools][biotools] (for biosciences) and [Awesome Research Software Registries][awesome-rs-registries], providing a list of research software registries (by country, organisation, domain and programming language) where research software can be registered to help promote its discovery
+
+### Persistent identifiers
+
+Unique persistent identifiers, such as Digital Object Identifiers (DOIs) provided by Zenodo, FigShare and similar digital archiving services, and commits/tags/releases used by GitHub and similar code sharing platforms, 
+help with findability and accessibility of your software, and can help you get credit for your work by providing citable references.
+
```

Comment:
I feel like there needs to be a section about CodeMeta here. It helps on F, I and R (especially on F)